### PR TITLE
Bugfix/login screen

### PR DESCRIPTION
**Pull request by simonschwartz**
This PR fixes the current styling defects in the admin login interface by applying the current visual supplied from the designers. This PR closes SITES-601 and SITES-583.

I have run these changes by @klepas from UI Kit team and we will hold off bringing this component into UI Kit as they aren't ready to start looking at form templates quite yet. There is also a high chance this design may change when there is more design thinking applied to the editorial experience.

CSS is annotated to make refactoring this stuff out later as painfree as possible :)

Before:
sign-in-old

New:
sign-in-new

### CODE REVIEW

File: app/views/devise/sessions/new.html.haml [hunk 1]
```diff
@@ -1,16 +1,41 @@
-%article.content-listing
-  %h2 Log in
-  %section.form-wrapper
+%article.sign-in
+  %h1 Sign in &mdash; GOV.AU Author
+  %section.sign-in-form
     = simple_form_for(resource, as: resource_name, url: session_path(resource_name)) do |f|
       / TODO: UI Kit needs styles for email type (see https://github.com/AusDTO/gov-au-ui-kit/issues/51)
```
Comment (maxious) [inline app/views/devise/sessions/new.html.haml]: I think this has been fixed https://github.com/AusDTO/gov-au-ui-kit/pull/52

File: app/views/devise/sessions/new.html.haml [hunk 2]
```diff
@@ -1,16 +1,41 @@
...
+        .form-options
+          .remember-me
+            = f.check_box :remember_me, as: :boolean if devise_mapping.rememberable?
+            %label.checkbox{ :for => "user_remember_me" } Remember me
+
+          %p
+            - if devise_mapping.recoverable? && controller_name != 'passwords' && controller_name != 'registrations'
+              = link_to "Forgot password?", new_password_path(resource_name)
+
+        / = f.input :remember_me, as: :boolean if devise_mapping.rememberable?
```
Comment (maxious) [inline app/views/devise/sessions/new.html.haml]: Commented out code not needed

File: app/views/devise/sessions/new.html.haml [hunk 3]
```diff
@@ -1,16 +1,41 @@
...
         = f.label :password
         = f.password_field :password
-      = f.input :remember_me, as: :boolean if devise_mapping.rememberable?
-      = f.button :submit, "Log in", role: 'button'
 
-  = render "devise/shared/links"
+      .form-footer-actions
+        .form-options
+          .remember-me
+            = f.check_box :remember_me, as: :boolean if devise_mapping.rememberable?
```
Comment (bjgordon) [inline app/views/devise/sessions/new.html.haml]: `if devise_mapping.rememberable?` should be on line 20

File: app/views/devise/sessions/new.html.haml [hunk 4]
```diff
@@ -1,16 +1,41 @@
-%article.content-listing
-  %h2 Log in
-  %section.form-wrapper
+%article.sign-in
+  %h1 Sign in &mdash; GOV.AU Author
```
Comment (bjgordon) [inline app/views/devise/sessions/new.html.haml]: Did this new title come from somewhere? I think it can just be "Sign in" without "GOV.AU Author". Author is just one role. Similar how the signup page just says "Sign up".

File: app/views/devise/sessions/new.html.haml [hunk 5]
```diff
@@ -1,16 +1,41 @@
...
+    .sign-in-footer-links
+      - if devise_mapping.confirmable? && controller_name != 'confirmations'
+        = link_to "Didn't receive confirmation instructions?", new_confirmation_path(resource_name)
+
+      %h2.background-border-heading
+        %span.inner
+          Or
+
+      - if devise_mapping.registerable? && controller_name != 'registrations'
+        = link_to "Sign up to be an author", new_registration_path(resource_name), :class => "primary-link"
```
Comment (bjgordon) [inline app/views/devise/sessions/new.html.haml]: At the risk of bike shedding, as above author sounds weird to me as they are just one role...

**Comment by klepas** [comment]
This is so hawt right now…

**Comment by bjgordon** [comment]
LGTM

File: app/views/devise/confirmations/new.html.haml [hunk 6]
```diff
@@ -1,10 +1,23 @@
...
+    .sign-in-footer-links
+
+      %h2.background-border-heading
+        %span.inner
+          Or
+
+      - if controller_name != 'sessions'
+        = link_to "Sign in", new_session_path(resource_name), :class => "primary-link"
+
+    / = render "devise/shared/links"
```
Comment (ltankey) [inline app/views/devise/confirmations/new.html.haml]: Can probably remove this too.

File: app/views/devise/passwords/new.html.haml [hunk 7]
```diff
@@ -1,9 +1,25 @@
...
+        = link_to "Didn't receive confirmation instructions?", new_confirmation_path(resource_name)
+
+      %h2.background-border-heading
+        %span.inner
+          Or
+
+      - if controller_name != 'sessions'
+        = link_to "Sign in", new_session_path(resource_name), :class => "primary-link"
+
+    / = render "devise/shared/links"
```
Comment (ltankey) [inline app/views/devise/passwords/new.html.haml]: Can probably remove this.

File: app/views/devise/passwords/new.html.haml [hunk 8]
```diff
@@ -1,9 +1,23 @@
...
+
+      %p
+        = f.input :email, required: true, autofocus: true
+      .form-footer-actions
+        %p
+          = f.button :submit, "Send me reset password instructions", role: 'button'
+
+    .sign-in-footer-links
+      - if devise_mapping.confirmable? && controller_name != 'confirmations'
+        = link_to "Didn't receive confirmation instructions?", new_confirmation_path(resource_name)
```
Comment (ltankey) [inline app/views/devise/passwords/new.html.haml]: Could we perhaps edit `/views/devise/shared/_links.html` and leave in the `= render "devise/shared/links"` call?

Comment (simonschwartz) [inline app/views/devise/passwords/new.html.haml]: @ltankey I had a look at how that may work. The issue is that the type of links changes depending on which page you are on. EG there 'Remember password' is unique to the Sign-in page, whereas the Sign-in link is redundant on the Sign-in page, but should appear on password rest and Sign-up page etc. 

TLDR: I believe the simplest way is to not use a `shared/links` call for now

**Comment by ltankey** [comment]
Sounds good @simonschwartz LGTM
